### PR TITLE
[C-2823] Improve autogenerated image logic

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Improves autogenerated image logic by only generating images if user hasnt already set their own playlist artwork.
Also improves edit playlist logic by generating image when if user decides to remove their own artwork. this prevents the playlist from being saved with no artwork.

Also updates the "generateImage" workers to simply return the first track artwork if there is only one track in the playlist.

### CODE REVIEW

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/src/components/fields/PickArtworkField.tsx [hunk 1]
```diff
@@ -87,23 +94,35 @@ export const PickArtworkField = (props: PickArtworkFieldProps) => {
...
-        onLoad={() => setIsLoading(false)}
+        onLoad={() =>
+          setStatus({ imageLoading: false, imageGenerating: false })
+        }
         style={styles.image}
         noSkeleton
       >
         <View style={styles.iconPicture}>
-          {isLoading ? (
+          {status.imageLoading ? (
```
Comment (Kyle-Shanks) [inline packages/mobile/src/components/fields/PickArtworkField.tsx:125]: does this cover the case of status.imageGenerating as well?

File: packages/web/src/workers/createPlaylistArtwork.worker.js
Comment (Kyle-Shanks) [inline packages/web/src/workers/createPlaylistArtwork.worker.js:1]: do we still need both of these workers? might have missed where one was used vs the other

**Review by Kyle-Shanks — APPROVED** [review]
awesome!

File: packages/mobile/src/components/fields/PickArtworkField.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/fields/PickArtworkField.tsx:125]: no great call

File: packages/web/src/workers/createPlaylistArtwork.worker.js
Comment (dylanjeffers) [reply packages/web/src/workers/createPlaylistArtwork.worker.js:1]: nice find, was not properly removed

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2823-fix-autogenerated-datum

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2823-fix-autogenerated-datum

**Pull request merged**